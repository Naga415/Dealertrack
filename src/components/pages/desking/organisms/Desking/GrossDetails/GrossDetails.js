'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Table from '../../../molecules/Table/Table';
import ProfitBreakdown from '../../../../../../models/ProfitBreakdown';
import { getFormattedNumber } from '../../../../../../utils/numberFormatter';

// eslint-disable-next-line react/prefer-stateless-function
class GrossDetails extends React.Component {
  render() {
    const { htmlId, frontGrossBreakdown, backGrossBreakdown } = this.props;
    return (
      <div id={this.props.htmlId}>
        <Table htmlId={`${htmlId}Front`} title="Front End Gross" dataFormatters={{ 'number': getFormattedNumber }} data={frontGrossBreakdown}
          columns={["Description", "Price", "Cost", "Profit"]}
          dataMap={{
            desc: "Description",
            amt: "Amount",
            cost: "Cost",
            profit: "profit"
          }} />
        <Table htmlId={`${htmlId}Back`} title="Back End Gross" dataFormatters={{ 'number': getFormattedNumber }} data={backGrossBreakdown}
          columns={["Description", "Price", "Cost", "Profit"]}
          dataMap={{
            desc: "Description",
            amt: "Amount",
            cost: "Cost",
            profit: "profit"
          }} />
      </div>
    );
  }
}

GrossDetails.propTypes = {
  htmlId: PropTypes.string,
  frontGrossBreakdown: PropTypes.arrayOf(PropTypes.instanceOf(ProfitBreakdown)),
  backGrossBreakdown: PropTypes.arrayOf(PropTypes.instanceOf(ProfitBreakdown))
};

export default GrossDetails;
