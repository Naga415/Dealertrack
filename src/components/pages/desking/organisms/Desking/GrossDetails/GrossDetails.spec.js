import React from 'react';
import renderer from 'react-test-renderer';
import GrossDetails from './GrossDetails';

describe('<GrossDetails/> component', () => {
  const frontGrossBreakdown = [

  ];

  const backGrossBreakdown = [

  ];

  it('renders <GrossDetails/> ', () => {
    const component = renderer.create(<GrossDetails htmlId="GrossDetails" frontGrossBreakdown={frontGrossBreakdown} backGrossBreakdown={backGrossBreakdown} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
