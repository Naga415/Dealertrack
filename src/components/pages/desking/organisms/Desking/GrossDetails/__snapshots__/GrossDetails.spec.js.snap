// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GrossDetails/> component renders <GrossDetails/>  1`] = `
<div
  id="GrossDetails"
>
  <div
    className="desking-table"
    id="GrossDetailsFront"
  >
    <span
      className="desking-table__title"
    >
      Front End Gross
    </span>
    <table>
      <thead>
        <tr
          className="table-headers"
        >
          <th>
            Description
          </th>
          <th>
            Price
          </th>
          <th>
            Cost
          </th>
          <th>
            Profit
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="desking-table"
    id="GrossDetailsBack"
  >
    <span
      className="desking-table__title"
    >
      Back End Gross
    </span>
    <table>
      <thead>
        <tr
          className="table-headers"
        >
          <th>
            Description
          </th>
          <th>
            Price
          </th>
          <th>
            Cost
          </th>
          <th>
            Profit
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
